patterns/request-reply/direct_replier_blocking.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"solace.dev/go/messaging"
+	"solace.dev/go/messaging/pkg/solace/config"
+	"solace.dev/go/messaging/pkg/solace/resource"
+)
+
+// getEnv function
+func getEnv(key, def string) string {
+	if val, ok := os.LookupEnv(key); ok {
+		return val
+	}
+	return def
+}
+
+func main() {
+	// logging.SetLogLevel(logging.LogLevelInfo)
+
+	// Configuration parameters
+	brokerConfig := config.ServicePropertyMap{
+		config.TransportLayerPropertyHost:                getEnv("SOLACE_HOST", "tcp://localhost:55555,tcp://localhost:55554"),
+		config.ServicePropertyVPNName:                    getEnv("SOLACE_VPN", "default"),
+		config.AuthenticationPropertySchemeBasicPassword: getEnv("SOLACE_PASSWORD", "default"),
+		config.AuthenticationPropertySchemeBasicUserName: getEnv("SOLACE_USERNAME", "default"),
+	}
+
+	messagingService, err := messaging.NewMessagingServiceBuilder().FromConfigurationProvider(brokerConfig).Build()
+
+	if err != nil {
+		panic(err)
+	}
+
+	// Connect to the messaging serice
+	if err := messagingService.Connect(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Connected to the broker? ", messagingService.IsConnected())
+
+	if err != nil {
+		panic(err)
+	}
+
+	topicSubscription := resource.TopicSubscriptionOf("solace/samples/*/direct/request")
+
+	//  Build a Request-Reply Message Receiver
+	requestReplyReceiver, builderErr := messagingService.
+		RequestReply().
+		CreateRequestReplyMessageReceiverBuilder().
+		Build(topicSubscription)
+
+	if builderErr != nil {
+		panic(builderErr)
+	}
+
+	// Start Request-Reply Message Receiver
+	if startErr := requestReplyReceiver.Start(); startErr != nil {
+		panic(startErr)
+	}
+
+	fmt.Println("Request-Reply Receiver running? ", requestReplyReceiver.IsRunning())
+
+	fmt.Println("\n===Interrupt (CTR+C) to handle graceful terminaltion of the subscriber===")
+
+	fmt.Println("Message Topic Subscription: ", topicSubscription.GetName())
+
+	// have receiver push request messages to request message handler
+	// Run in a seperate Go routine
+	var receivedMsgCounter = 0 // counter for the received messages
+
+	// Run forever until an interrupt signal is received
+	for requestReplyReceiver.IsRunning() {
+		// have receiver push request messages to request message handler
+		message, replier, regErr := requestReplyReceiver.ReceiveMessage(-1)
```

Comment:
Can we please put a comment or a reference to the API doc to why `-1` is added as the function parameter to this ReceiveMessage? Thanks!

Reply:
@TamimiGitHub, this is the reference to the API doc - https://pkg.go.dev/solace.dev/go/messaging@v1.6.1/pkg/solace#RequestReplyMessageReceiver . I can add it in a separate PR if you want me to

Reply:
Could you add it to the same PR please and we can merge right away?

Reply:
I have added it in the provision/deprovision PR - #13